base-isa.md

```diff
@@ -19,50 +20,62 @@ The highest two bits of the first byte are a format marker:
+| `10 1 ?????`  | `?????????`  | reserved for extensions                          |
+| `11 ??????`   | `?????????`  | reserved for extensions                          |
+
+| Symbol | Meaning         |
+|--------|-----------------|
+| C      | Opcode          |
+| R      | Register ID     |
+| I      | Immediate       |
+| D      | Displacement    |
+| S      | Operation Size  |
+| M      | Memory Addressing Mode     |
+| ?      | Arbitrary Value |
+
+Notice that whenever `MM` bits are present, it is reserved for their value to be anything other than `00`. Similarly, it is reserved for `SS` bits to have values other than `01`.
+
+### Execution of Illegal and Reserved Instructions
+
+Execution of an _illegal_ or _reserved_ instruction is handled through an implementation specific method.
 
 ## Computation Instructions
 
-Both computation formats share a lot of similarities. For both the first byte has the structure `0x SS CCCC`.
+Both computation formats share a lot of similarities. For both, the first byte has the structure `0x SS CCCC`.
 
-- `SS` is a size marker and reserved for extensions. For the base instructions this should always be `01`
-- `CCCC` is a 4bit opcode deciding which operation to execute. This can be an ALU or a memory load/store instruction
+- `SS` is a size marker and reserved for extensions. For the base instructions this is always `01`.
+- `CCCC` is a 4 bit opcode deciding which operation to execute.
 
 ### 2 Register Computation
 
-The second byte has the format `AAA BBB 00` where `AAA` and `BBB` are references to registers. `AAA` is the destination register and left source register. `BBB` is the right source register. The only exception is the RSUB instruction where `AAA` and `BBB` are swapped for the purpose of being source registers.
+The second byte has the format `AAA BBB MM` where `AAA` and `BBB` are references to registers. `AAA` is the destination register and left source register. `BBB` is the right source register.
 
 ### Immediate Computation
 
-The second byte has the format `RRR IIIII`, where the 5bit immediate acts as operand `B`. The immediate is sign extended for operations 0-7 and operation 9. The immediate is zero extended for operation 8 and operations 10-15. While interleaving of signed and unsigned immediate at the border of the 2 sections seems unusual at first glance, it's done to make decoding hardware simpler.
+The second byte has the format `AAA IIIII`, where the 5 bit immediate acts as operand `B`. The immediate is sign extended for operations 0-7 and operation 9. The immediate is zero extended for operation 8 and operations 10-15. `AAA` is treated the same as in the 2 Register Computation section above.
 
-### Opcodes
+### Exceptions
 
-TODO: This is a baseline, very much still floating
+There are several exceptions to how the operands work.
+
+ - `RSUB`
+    - `BBB` is the left source register and `AAA` is the right source register.
```

Comment:
Should it be mentioned that this also applies to Immediate mode? (So `IIIII` or just `B` instead of `BBB`?) or maybe it's ok like this, IDK.

Reply:
Good catch!